packages/app/src/vis-packs/core/utils.ts

```diff
@@ -37,17 +37,20 @@ export function applyMapping<T>(
     return undefined;
   }
 
+  const slicingState = mapping.map((val) => (isNumber(val) ? val : null));
+  const slicedView = slicingState.some((val) => val === null)
+    ? baseArray.pick(...slicingState)
+    : baseArray; // no extra dimension to slice, or back-end has already done it
+
   const isXBeforeY =
     mapping.includes('y') &&
     mapping.includes('x') &&
     mapping.indexOf('x') < mapping.indexOf('y');
 
-  const slicingState = mapping.map((val) => (isNumber(val) ? val : null));
```

Comment:
Hm, what about putting an early return of `baseArray` here if `slicingState.every((val) => val === null) && !isXBeforeY` ?

This would remove the need for the two ternaries for `slicedView` and for the return value.